drivers/hmis/app/models/hmis/project_group.rb

```diff
@@ -0,0 +1,110 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Hmis
+  class ProjectGroup < GrdaWarehouseBase
+    self.table_name = 'hmis_project_groups'
+    acts_as_paranoid
+    has_paper_trail
+    include ::Hmis::Concerns::HmisArelHelper
+
+    validate :data_source_must_be_hmis
+
+    belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+    has_and_belongs_to_many :projects, class_name: 'Hmis::Hud::Project', join_table: :hmis_project_project_groups, foreign_key: :hmis_project_group_id
+    has_and_belongs_to_many :warehouse_projects, class_name: 'GrdaWarehouse::Hud::Project', join_table: :hmis_project_project_groups, foreign_key: :hmis_project_group_id
+
+    # GroupViewableEntities that are associated with this project group
+    has_many :hmis_group_viewable_entities, -> { where(entity_type: 'Hmis::ProjectGroup') }, class_name: 'Hmis::GroupViewableEntity', foreign_key: :entity_id
+
+    scope :viewable_by, ->(user) do
+      # Only HMIS admins. We may expand visibility of project groups later with new permissions.
+      return all if HmisEnforcement.hmis_admin_visible?(user)
+
+      none
+    end
+
+    scope :editable_by, ->(user) do
+      # Only HMIS admins. We may expand visibility of project groups later with new permissions.
+      return all if HmisEnforcement.hmis_admin_visible?(user)
+
+      none
+    end
+
+    # Search for project groups by group name or project name
+    scope :text_search, ->(text) do
+      query = text.gsub(/[^0-9a-zA-Z ]/, '')
+      return none unless query.present?
+
+      distinct.left_outer_joins(:projects).
+        where(
+          arel_table[:name].lower.matches("%#{query.downcase}%").
+          or(p_t[:ProjectName].lower.matches("%#{query.downcase}%")),
+        )
+    end
+
+    def any_inclusion_criteria?
+      JSON.parse(inclusion_criteria).compact_blank.any?
+    end
+
+    def any_exclusion_criteria?
+      JSON.parse(exclusion_criteria || '{}').compact_blank.any?
+    end
```

Comment:
Should `any_inclusion_criteria?` also:
```suggestion
    def any_inclusion_criteria?
      JSON.parse(inclusion_criteria || '{}').compact_blank.any?
    end

    def any_exclusion_criteria?
      JSON.parse(exclusion_criteria || '{}').compact_blank.any?
    end
```

Or is that safe since we set it on create?  Maybe we should set `exclusion_criteria` to an empty hash on create as well, or make that the default for both columns?

Reply:
I made the `inclusion_criteria` column non-nullable in the database, so we shouldn't need that. But in practice now with the view, they both always get set to something (whether that's `"{}"` or an object with contents).